src/Cli/Microsoft.DotNet.Cli.Utils/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.DotNet.Cli.Utils.Extensions;
+
+public static class TypeExtensions
+{
+    // This is used when outputting the Type information for the CLI schema JSON.
+    public static string ToCliTypeString(this Type type)
```

Comment:
This transforms the normal CLR type string into a more human-readable string for the CLI schema to output.